packages/core/src/BinaryDataManager/LocalStorage.ts

```diff
@@ -0,0 +1,121 @@
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { IBinaryData } from 'n8n-workflow';
+import { v4 as uuid } from 'uuid';
+
+import { IBinaryDataConfig, IBinaryDataManager } from '../Interfaces';
+
+export class BinaryDataLocalStorage implements IBinaryDataManager {
+	private storagePath = '';
+
+	private binaryDataTTL = 60;
+
+	async init(config: IBinaryDataConfig, startPurger = false): Promise<void> {
+		this.storagePath = config.localStoragePath;
+		this.binaryDataTTL = config.binaryDataTTL;
+		if (startPurger) {
+			setInterval(async () => {
+				// get all files and delete data
+				await this.deleteMarkedFiles();
+			}, this.binaryDataTTL * 30000);
+		}
+
+		return fs
+			.readdir(this.storagePath)
+			.catch(async () => fs.mkdir(this.storagePath, { recursive: true }))
+			.then(async () => fs.readdir(this.getBinaryDataMetaPath()))
+			.catch(async () => fs.mkdir(this.getBinaryDataMetaPath(), { recursive: true }))
+			.then(() => {});
+	}
+
+	async storeBinaryData(binaryData: IBinaryData, binaryBuffer: Buffer): Promise<IBinaryData> {
+		const retBinaryData = binaryData;
+		retBinaryData.internalIdentifier = uuid();
+
+		return this.saveToLocalStorage(binaryBuffer, retBinaryData.internalIdentifier).then(
+			() => retBinaryData,
+		);
+	}
+
+	async retrieveBinaryDataByIdentifier(identifier: string): Promise<Buffer> {
+		return this.retrieveFromLocalStorage(identifier);
+	}
+
+	async markDataForDeletion(identifiers: string[]): Promise<void> {
```

Comment:
Something strange is happening. On my machine I have a simple workflow that reads the same file 26 times (I used a `Function` node to generate 26 empty outputs).

This results correctly in 26 files being created in the `binaryData` folder.

The problem I'm facing is that the meta file remains empty and even after the executions, the files remain there.

Have you experienced this? This seems to be a possible storage problem.

Reply:
After clarifications I understand that executions are saved forever unless executions are not supposed to be saved (like manual executions). In this case, they'll be deleted upon cron execution.